k-distribution/src/main/scripts/bin/kparse

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+set -u
+
+export PATH="$(cd "$(dirname "$0")"; pwd):$PATH"
+
+# initialize flags
+ARGV=()
+dir=.
+fold_lines="fold -s"
+inputMode=program
+literal=false
+outputFile=-
+result=1
+verbose=false
+
+# setup temp files
+now=`date +"%Y-%m-%d-%H-%M-%S"`
+tempDir="$(mktemp -d .kparse-${now}-XXXXXXXXXX)"
+tempFiles=("$tempDir")
+trap 'rm -rf ${tempFiles[*]}' INT TERM EXIT
+
+execute () {
+  (
+  if $verbose; then
+    set -x
+  fi
+  "$@"
+  )
+}
+
+KPARSE=$(basename "$0")
+KAST_UTIL="$(dirname "$0")/kast"
+
+error () {
+  printf "[Error] Inner Parser: $@\n" | $fold_lines 1>&2
+  exit $result
+}
+
+print_usage () {
+cat <<HERE
+Usage: $KPARSE <INPUT_FILE> [-d DIR] [-m MODULE] [-s SORT]
+Generate an on-the-fly parser from a K definition and use it to parse a term.
+
+Mandatory arguments to long options are mandatory for short options too.
+
+$KPARSE expects exactly 1 positional argument. It must be either a readable
+file, or '-'. By default, this file will be parsed using the main syntax
+module of the K definition and the sort of the \$PGM configuration variable.
+The resulting AST, in KORE format, will be output on stdout. This can be
+customized; see below. If no \$PGM configuration variable exists, the default
+sort will be the KItem sort.
+
+  -d, --directory DIR      Look for a kompiled directory ending in "-kompiled"
+                           under the directory DIR.
+  -h, --help               Display this help and exit
+  -i, --input MODE         Select input mode to use when parsing. Valid values
+                           are program, kast, binary, json, and kore. 
+  -m, --module MODULE      Module to use to generate grammar
+      --no-exc-wrap        Do not wrap exception messages to 80 chars. Keep
+                           long lines.
+      --output-file FILE   Print resulting AST to FILE
+      --save-temps         Do not delete temporary files when $KPARSE
+                           terminates.
+  -s, --sort               Sort to use as start symbol of parser
+  -v, --verbose            Print significant commands executed by $KPARSE.
+      --version            Print version of K
+      --                   Interpret all arguments following as positional
+                           arguments.
```

Comment:
Periods on sentences.

Reply:
I couldn't tell if you were saying that there should be or should not be periods, but I looked at other --help messages for existing linux tools and they seem to generally not use periods, so I've made it consistent that way.